### ISSUE TITLE
training techniques you suggested

### ISSUE THREAD

**Issue by min6k** [arcaninez/ODM3D#2]
I would like to reproduce the performance of paper using the kitti-3D dataset you tested. 
But there are difficulties in reproducing the performance of your KITTI VALIDATION set. 

Settings in preferences (* CUDA 11.4 / python == 3.8 / pytorch == 1.10.1) 
I followed the training techniques you suggested. 30 epochs were learned in 1STATGE and 15 epochs in 2STAGE. 

scd-teacher was used in the same way. The checkpoint size after 1st stage learning is as follows.
![image](https://github.com/user-attachments/assets/7ec4e015-0c46-40e7-b393-fc2ce2f87b7c)

Stage 2 is as follows.
![1](https://github.com/user-attachments/assets/4ef24cee-f81a-4665-932d-ffff321c6642)

training set & validation set 
'train': [kitti_infos_kitti_train_and_eigen_clean_soft.pkl],
'val': [kitti_infos_val.pkl],

However, the performance is low by 2%. You have not modified your GITHUB code. I want to find the cause. Training was conducted with a larger epoch, but the performance was low. Please let me know what I need to modify to reproduce performance.

**Comment by arcanienz** [comment]
Hi, thanks for your interest in our work.

The provided code shall easily get even higher results than the ones reported in the paper. I'd suggest you to try removing CMAug, because this should still give you something close to paper reported results. 

Also, you could keep ablating the designs of ODM3D towards the CMKD baseline (you should also be able to reproduce the results of CMKD). Simply applying the proposed BEV feature distillation is supposed to give results no less than 2% of the reported results.

The code provided in this repo should provide results on par with if not better than the paper reported ones, as have been verified by myself and other users.

Best.

**Comment by min6k** [comment]
Thank you for your reply.

I have one more question for you.
When setting up a training load dataset, I was wondering if I should change 'DATA_SPLIT' in the 'kitti_dataset.yaml' file?


1. i don't know if it's correct to have 'DATA_SPLIT' as 'train' in the following image. I don't know if it's correct to have 'kitti_train_and_eigen_clean'.

2.'DB_INFO_PATH:' in 'DATA_AUGMENTOR' should be 'kitti_dbinfos_train.pkl'? (only train set? not eigen_clean?)

I look forward to your reply. 

-----Original Message-----

**Comment by xiaosenho** [comment]
Hello, could you please provide the evaluation metrics of the validation set results you reproduced?

**Comment by min6k** [comment]
I used kitti eval metric from your github.
I measured performance based on AP_R40 metric in kitti.


Here is eval.py. When I checked, there was no modification in your eval code.

(ODM3D/pcdet/datasets/kitti/kitti_object_eval_python/eval.py)

-----Original Message-----

**Comment by xiaosenho** [comment]
So what is your result? I am also trying to reproduce the experiment

**Comment by arcanienz** [comment]
@min6k Hi, sorry for the late response. As I'm away from office, I'll check these details and try to get back to you within a week.

**Comment by arcanienz** [comment]
@min6k Hi, 

For 1, in the training scripts, only odm3d_s1 and odm3d_s2 are required. kitti_dataset.yaml is used when generating the required pkl data files. For the generation of kitti_infos_val.pkl, you may refer to the guide in CMKD's GETTING_STARTED.md (in which I think  'train': train and  'test': val are used).

For 2, it should be 'kitti_dbinfos_train.pkl' because it is a database that contains all annotated objects in the training set. When training on the entire training and validation set, this should be 'kitti_dbinfos_trainval.pkl'. Please see the "CMAug Workflow" paragraph in the paper.

Since this work was completed quite a while ago, I only vaguely remember some of the details. Feel free to discuss more.

Best.
